src/socket/socket.js

```diff
@@ -86,19 +84,59 @@ export function initializeSocket(httpServer) {
         return;
       }
       // Example usage of modifyPoints function
-     
+
       const leaderboard = generateLeaderboard(roomId);
       io.to(socket.id).emit("leaderboard", leaderboard);
     });
 
+    let imageUploadOrder = [];
+
+    // Modify the socket.on function to keep track of the image upload order
     socket.on("upload", (data) => {
       console.log("got an image");
       const base64Data = data.image.replace(/^data:image\/\w+;base64,/, "");
       const buffer = Buffer.from(base64Data, "base64");
       const filename = Date.now() + "-" + data.filename;
       const isCorrectObject = checkImage(filename, buffer, "bottle");
       console.log(isCorrectObject);
+
+      // Add the username to the image upload order
+      imageUploadOrder.push(data.username);
```

Comment:
data object does not have username property. (i should've also made a check condition if the data incoming is in correct format or not. will fix it in the next commit)

also, its better to identify user with their socket id because its truly unique and can't be same for 2 users

Reply:
i have look it